middleware/index.ts

```diff
@@ -184,13 +200,29 @@ export default function applySfFxMiddleware(request: any, state: any, resultArgs
         throw new Error('Request Data not provided');
     }
 
+    // Logger is the last element in the array
+    const logger: Logger = resultArgs[resultArgs.length-1];
+    if (!logger) {
+        throw new Error('Logger not provided in resultArgs from node system function');
+    }
+
+    //use secret her in lieu of DEBUG runtime environment var until we haave deployment time support of config var
+    const debugSecret = getSecret("sf-debug", "DEBUG", logger);
+    logger.info(`DEBUG flag is ${debugSecret}`);
+    if(debugSecret || LoggerLevel.DEBUG === logger.getLevel() || process.env.DEBUG) {
+        //for dev preview, we log the ENTIRE raw request, may need to filter sensitive properties out later
+        //the hard part of filtering to know which name to filter
+        let childLogger: Logger = logger.child('raw_request', request);
+        childLogger.info('debug raw request in middleware');
+    }
+
     const data = request.payload.data;
     if (!data) {
-        throw new Error('Data not provided');
+        throw new Error('Data field of the cloudEvent not provided in the request');
     }
 
     if (!data.context) {
-        throw new Error('Context not provided in data');
+        logger.warn('Context not provided in data, context.org is not initialized');
```

Comment:
It's actually more than `context.org`.  Can we say ": context partially initialize"?